modules/raft/src/integrationTest/java/org/apache/ignite/raft/jraft/core/ItNodeTest.java

```diff
@@ -4497,6 +4509,36 @@ private static boolean waitForTopology(TestCluster cluster, PeerId peerId, int e
         return false;
     }
 
+    /**
+     * Test if the actual value gets equal to expected within a given timeout. The method will log all values the actual parameter
+     * has taken if the timeout is reached.
+     * @param expected the expected value
+     * @param actual the actual value, this should change over time and therefore a supplier
+     * @param timeout the duration within the actual value should be the same as the expected value
+     * @param <T> the type of expected and actual value
+     * @throws TimeoutException when the duration is reached
+     */
+    @SuppressWarnings("BusyWait")
+    private static <T> void assertWaitForCondition(T expected, Supplier<T> actual, Duration timeout) throws TimeoutException {
```

Comment:
While I understand that this method can be helpful, I think it could be written in a simpler way, reusing the existing method:

```
boolean success = waitForCondition(() -> {
    T actualVal = actual.get();

    if (Objects.equals(expected, actualVal)) {
        return true;
    } else {
        //no matching result, save for debug
        results.add(actualVal);
        return false;
    }
}, timeout.toMillis());

if (!success) {
    fail(String.format("Timeout reached while waiting for expected result. Expected: %s, Actual Results: %s", expected, results));
}
```

Reply:
Yes, of course I edited it.